src/app/settings/edit-account/page.tsx

```diff
@@ -6,16 +6,121 @@ import ActiveButton from '@/components/Active-Button/ActiveButton';
 import Image from 'next/image';
 import backIcon from '@/assets/BackArrow.svg';
 import Link from 'next/link';
+import { Bounce, toast, ToastContainer } from 'react-toastify';
+import useDarkMode from '@/hooks/useDarkMode';
+import 'react-toastify/dist/ReactToastify.css';
+import { validateEmail } from '@/lib/fieldValidator';
+import { updateUserEmail } from '@/hooks/apis/users';
+import { canParseJson } from '@/utils/helpers';
+import { generateTokenAndEmail } from '@/lib/generateVerification';
+import { useRouter } from 'next/navigation';
 
 const Editaccount: React.FC = () => {
-  const userEmail = 'Deerbuttcheeks@gmail.com';
+  const userEmail = 'new@example.com'; // Replace with actual user email from authentication state or context
   const [email, setEmail] = useState(userEmail);
-  const [isEmailMatch, setIsEmailMatch] = useState(false);
+  const [isEmailMatch, setIsEmailChanged] = useState(false);
+  const [emailError, setEmailError] = useState('');
+  const isDarkMode = useDarkMode();
+  const router = useRouter();
 
   const handleEmailChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     const value = e.target.value;
     setEmail(value);
-    setIsEmailMatch(value !== userEmail);
+    setIsEmailChanged(value !== userEmail);
+    setEmailError('');
+  };
+
+  const handleSave = async () => {
+    const emailErr = validateEmail(email);
+    if (emailErr) {
+      setEmailError(emailErr);
+      return;
+    }
+
+    try {
+      const response = await updateUserEmail(userEmail, email);
```

Comment:
rename these variables so its more clear on which is the old email and which is the new one

Reply:
updated, good catch